pkg/cluster/nodes.go

```diff
@@ -0,0 +1,56 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// NodeImages returns a map of node names to maps of images present on that node
+func NodeImages(ctx context.Context, client kubernetes.Interface) (map[string]map[string]struct{}, error) {
+	nodes, err := client.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("unable to list nodes: %w", err)
+	}
+
+	nodeImages := map[string]map[string]struct{}{}
+
+	for _, node := range nodes.Items {
+		thisNodeImages := map[string]struct{}{}
+		for _, image := range node.Status.Images {
```

Comment:
I'm pretty sure that node.Status.Images is limited to first 50 so this will be unreliable.

Reply:
Ouch, yes it will be
I don't think there's a good solution to that though (besides making sure we don't ask for the images to be loaded again on the node the script is being run from)